encoding/protobuf/error_integration_test.go

```diff
@@ -0,0 +1,185 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package protobuf_test
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/gogo/googleapis/google/rpc"
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/gogo/status"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/encoding/protobuf"
+	"go.uber.org/yarpc/encoding/protobuf/internal/testpb"
+	"go.uber.org/yarpc/encoding/raw"
+	"go.uber.org/yarpc/transport/grpc"
+	"go.uber.org/yarpc/yarpcerrors"
+)
+
+type errorServer struct{}
+
+func (errorServer) Unary(ctx context.Context, msg *testpb.TestMessage) (*testpb.TestMessage, error) {
+	testDetails := []proto.Message{
+		&types.StringValue{Value: "string value"},
+		&types.Int32Value{Value: 100},
+	}
+	return nil,
+		protobuf.NewError(yarpcerrors.CodeInvalidArgument, msg.Value,
+			protobuf.WithErrorDetails(testDetails...))
+}
+
+func (errorServer) Duplex(stream testpb.TestServiceDuplexYARPCServer) error {
+	return nil
+}
+
+func TestProtoGrpcServerErrorDetails(t *testing.T) {
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+
+	inbound := grpc.NewTransport().NewInbound(listener)
+	dispatcher := yarpc.NewDispatcher(yarpc.Config{
+		Name:     _serverName,
+		Inbounds: yarpc.Inbounds{inbound},
+		Logging:  yarpc.LoggingConfig{},
+		Metrics:  yarpc.MetricsConfig{},
+	})
+
+	dispatcher.Register(testpb.BuildTestYARPCProcedures(&errorServer{}))
+	require.NoError(t, dispatcher.Start(), "could not start server dispatcher")
+
+	addr := inbound.Addr().String()
+	outbound := grpc.NewTransport().NewSingleOutbound(addr)
+	clientDispatcher := yarpc.NewDispatcher(yarpc.Config{
+		Name: _clientName,
+		Outbounds: map[string]transport.Outbounds{
+			_serverName: {
+				ServiceName: _serverName,
+				Unary:       outbound,
+				Stream:      outbound,
+			},
+		},
+		Logging: yarpc.LoggingConfig{},
+		Metrics: yarpc.MetricsConfig{},
+	})
+
+	client := testpb.NewTestYARPCClient(clientDispatcher.ClientConfig(_serverName))
+	require.NoError(t, clientDispatcher.Start(), "could not start client dispatcher")
+
+	defer func() {
+		assert.NoError(t, dispatcher.Stop(), "could not stop dispatcher")
+		assert.NoError(t, clientDispatcher.Stop(), "could not stop client dispatcher")
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	_, err = client.Unary(ctx, &testpb.TestMessage{Value: "error msg"})
```

Comment:
let's check the response as well - it should be nil

Reply:
In case of error, proto encoding does not return empty response 
https://github.com/yarpc/yarpc-go/blob/ec8187ef475cef148e92b9f8737640557e0978e4/encoding/protobuf/outbound.go#L97